### PR TITLE
[AKS] Fix for `az aks approuting update` command

### PR DESCRIPTION
**Pull request by bosesuneha**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az aks approuting update -g 'resource_group' -n 'cluster_name' --enable-kv --attach-kv 'keyvault_id'

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/aks-preview/azext_aks_preview/custom.py [hunk 1]
```diff
@@ -2864,6 +2864,7 @@ def _keyvault_update(
             raise InvalidArgumentValueError("Please provide a valid keyvault ID")
 
         if mc.ingress_profile and mc.ingress_profile.web_app_routing and mc.ingress_profile.web_app_routing.enabled:
+            cmd.command_kwargs['operation_group'] = 'vaults'
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/custom.py:2867]: Cannot tell what the change is trying to fix. Without mutating the value of `operation_group`, would keyvault_client.get fail?

Besides, it seems that the existing test case test_aks_approuting_update does not cover the scenario to update key vault, would you help update the test case?

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 2]
```diff
@@ -8508,6 +8508,7 @@ def test_aks_approuting_enable_with_keyvault_secrets_provider_addon(self, resour
         delete_cmd = 'aks delete --resource-group={resource_group} --name={aks_name} --yes --no-wait'
         self.cmd(delete_cmd, checks=[self.is_empty()])
 
+    @live_only()
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:8511]: Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=84038786&view=results) to validate the change, test passed!

Comment (FumingZhang) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:8511]: Discussed offline, as the role assignment operations could not be recorded and replayed correctly, marked the case as live only. Better left a comment here

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**